experiments/dqa/scripts/preprocess_diagram_annotations.py

```diff
@@ -3,18 +3,39 @@
 import sys
 import ujson as json
 import random
+import re
 
 diagram_file = sys.argv[1]
-out_file = sys.argv[2]
+diagram_size_file = sys.argv[2]
+out_file = sys.argv[3]
 text_labels = ["A", "B", "C", "D", "E", "F"]
 
+diagram_sizes = {}
+with open(diagram_size_file, 'r') as f:
+    lines = f.readlines()
+
+    for ind in xrange(len(lines) / 3):
+        i = ind * 3
+        idline = lines[i]
+        id = idline[(idline.rfind('/') + 1):].strip()
+        height = re.search("[0-9]+", lines[i + 1].strip()).group(0)
+        width = re.search("[0-9]+", lines[i + 2].strip()).group(0)
+
+        # print id, width, height
+        diagram_sizes[id] = (int(width), int(height))
+    
+
 output = []
 with open(diagram_file, 'r') as f:
     j = json.load(f)
 
     for t in j.iterkeys():
         diagrams = j[t]
         for diagram_id in diagrams.iterkeys():
+            if not diagram_sizes.has_key(diagram_id):
```

Comment:
`if not diagram_id in diagram_sizes` would be more python (and would work in Python 3)